lib/kitchen/id_tracker.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module Kitchen
+  # A class to track and modify duplicate IDs in a document
+  #
+  class IdTracker
+
+    def initialize
+      @id_cp_count = Hash.new { |hash, key| hash[key] = { count: 0, last_pasted: false } }
```

Comment:
this variable is more than a "copy count" now - suggest renaming to something like `@id_data`